### ISSUE TITLE
Inefficient include method

### ISSUE THREAD

**Issue by GoogleCodeExporter** [roundcubevnz/myroundcube#330]
```
I've been doing some performance profiling of our Roundcube installation and 
noticed a lot of lstat calls for files relating to the calendar plugin. The 
reason for this is that you are using relative paths for your includes, and 
there are several paths in the include path before ".". This obviously isn't a 
major issue, but on large installations with NFS storage (such as ours), it can 
have a measurable impact on performance. 

Given that the exact location of your includes is known, I would suggest using 
a full path instead. For the time being, I have replaced includes such as 
this...

require_once('program/class.http.php')

...with this...

require_once(__DIR__ . '/program/class.http.php')

Please would you consider making a similar change in the standard release?

Thanks

Marcus
```

Original issue reported on code.google.com by `m...@names.co.uk` on 2 May 2012 at 9:24

**Comment by GoogleCodeExporter** [comment]
```
The is a Roundcube constant INSTALL_PATH. So this should be fine as well:

require_once(INSTALL_PATH . 'plugins/calendar/program/ical.class.php');

Regarding the standard release, please mailto dev@lists.roundcube.net or open a 
ticket @ http://trac.roundcube.net.
```

Original comment by `rol...@liebl.ath.cx` on 2 May 2012 at 8:15
- Changed state: **Accepted**

**Comment by GoogleCodeExporter** [comment]
```
Fixed in calendar v8.8
```

Original comment by `rol...@liebl.ath.cx` on 10 May 2012 at 7:50
- Changed state: **Fixed**
